data-service/README.md

```diff
@@ -15,9 +15,48 @@ limitations under the License.
 --->
 # Data Service
 
-The core API for the Data Service accepts a request from the client and returns the resources that have been processed from the initial request for resources.  These resources will be filtered and redacted based on the processing of the request in the inital resource request.  
+The Data Service accepts client requests to retrieve resources that have been registered on their behalf. These will be
+the response to the initial requests sent to the Palisade service that have been collected, filtered and possibly
+redacted in conformance to the defined rules and to the context of the request. The client is expected to send a
+request containing the token and resource id that is used to uniquely identify the resource request. The response will
+be an output stream holding the data resources. To see more information on client requests, see the Palisade Client's
+library.
+
+The key components of the service is the implementation of the DataService interface and the supporting services.  This
+comes in the form of an implementation of DataService interface, SimpleDataService, that uses the Palisade Reader
+library for the implementation of a solution using a database. This is then wrapped in a class, AuditableDataService,
+that will provide the data in a form that can be use in constructing the response that is sent back to the client, and
+the audit message that is sent to the Audit Service using the AuditMessageService.
+
+
+```java
+interface DataService extends Service { {
+  CompletableFuture<AuthorisedDataRequest> authoriseRequest(final DataRequest request);
+
+  CompletableFuture<Boolean> read(final AuthorisedDataRequest request, final OutputStream out, final AtomicLong recordsProcessed, final AtomicLong recordsReturned);
+
+
+}
+```
+
+
+# Flow of Control
+
+o![Data Service diagram](doc/data-service.png)
```

Comment:
```suggestion
![Data Service diagram](doc/data-service.png)
```
The flow image should be colour coordinated with kafka sources being purple, for example. A good example of the colours in use can be seen in the [FRS flow diagram](https://github.com/gchq/Palisade-services/blob/develop/filtered-resource-service/doc/filtered-resource-service.png).
The lines in the diagram are not easy to see so it might be worth while looking at what I've done for the [User Service](https://github.com/gchq/Palisade-services/blob/Pal-872-user-service-streaming-documentation/user-service/doc/user-service.png)